### DISCUSSION TITLE
Organization of Bob Project

### DISCUSSION THREAD

**Discussion by beckhamrryyaann** [IBM/ibmi-tobi#140] [General]
I have an Accounting Application.  It is logically separated into 3 distinct sub-applications.  
1. Accounts Receivable 2. Payables 3. General Ledger.   I would like to have a root project called accounting.  Then have subdirectories for each sub app?   Can Bob handle this nesting of app structure?  When I get to my sub app I then need to have my source compile to one library and my tables built in another using the ibmi.json config files to control this.

Example Project structure:
accounting
-> AR
---->qrpglesrc
------->ibm.json
---->qddlsrc
------->ibmi.json
->AP
---->qrpglesrc
---->qddlsrc
etc.

**Comment by edmundreinhardt** [comment]
yes.   Bob is very flexible, you can have your source in an arbitrary directory structure.
You simply have a Rules.mk in every directory that has compilable source.  They will all be gathered dynamically when you invoke makei.

**Reply by beckhamrryyaann** [reply]
Just to confirm I was able to get this to work as intended.  Thanks for confirmation.

**Reply by edmundreinhardt** [reply]
Awesome!
